### PR TITLE
Make `ln-dlc-node` tests a bit more resilient

### PR DESCRIPTION
**Pull request by luckysori**
This was some work that I've been collecting as I've been trying to verify that the LN channel is usable after a DLC channel has been opened beside it.

### CODE REVIEW

File: crates/ln-dlc-node/src/tests/just_in_time_channel.rs [hunk 1]
```diff
@@ -22,17 +22,17 @@ async fn just_in_time_channel() {
...
+    let coordinator_outbound_liquidity_sat =
+        min_outbound_liquidity_channel_creator(&payer, payer_outbound_liquidity_sat);
+
     let payer_coordinator_channel_details = coordinator
         .open_channel(
-            &payer.info,
+            &payer,
             coordinator_outbound_liquidity_sat,
-            payer_inbound_liquidity_sat,
+            payer_outbound_liquidity_sat,
```
Comment (holzeis) [inline crates/ln-dlc-node/src/tests/just_in_time_channel.rs:35]: 🙃 depends on the point of view 😜

File: crates/ln-dlc-node/src/tests/mod.rs [hunk 2]
```diff
@@ -192,6 +204,40 @@ async fn fund_and_mine(address: Address, amount: Amount) -> Result<()> {
...
 
+/// Calculate the "minimum" acceptable value for the outbound liquidity
+/// of the channel creator.
+///
+/// The value calculated is not guaranteed to be the exact minimum,
+/// but it should be close enough.
+///
+/// This is useful when the channel creator wants to push as many
+/// coins as possible to their peer on channel creation.
+fn min_outbound_liquidity_channel_creator(peer: &Node, peer_balance: u64) -> u64 {
```
Comment (holzeis) [inline crates/ln-dlc-node/src/tests/mod.rs:215]: :+1: very good.. I think we will need something like this as a validation function on the frontend as well. I tested to open a position with a very small amount and run into the reserve issue.

**Review by holzeis — APPROVED** [review]
Nice!

File: crates/ln-dlc-node/src/tests/just_in_time_channel.rs (hunk 1 above)
Comment (luckysori) [reply crates/ln-dlc-node/src/tests/just_in_time_channel.rs:35]: The thing is the coordinator's outbound liquidity is the payer's inbound liquidity, so we can't use both variables together.

File: crates/ln-dlc-node/src/tests/mod.rs (hunk 2 above)
Comment (luckysori) [reply crates/ln-dlc-node/src/tests/mod.rs:215]: That is true, we can apply a similar logic when opening any channel.

What channel were you thinking off specifically. I suppose the coordinator-app channel.

Comment (holzeis) [reply crates/ln-dlc-node/src/tests/mod.rs:215]: I was referring to the dlc channel.

**Pull request merged**